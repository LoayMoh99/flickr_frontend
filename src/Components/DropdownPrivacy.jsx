/* eslint-disable linebreak-style */
import React from 'react';
import './DropdownPrivacy.css';

function DropdownPrivacy() {
  return (
    <>
      {/* <div className="styled"> */}

      {/* <select className="form-control" data-style="btn-primary"> */}
      <select>
        <span className="styled" />
        <option value="Public">Public</option>
        <option value="Private">Private</option>
        <option value="Family">Family</option>
        <option value="Friends">Friends</option>
        <option value="Friends & Family">Friends & Family</option>
      </select>

      {/* </div> */}
    </>
  );
}

export default DropdownPrivacy;
