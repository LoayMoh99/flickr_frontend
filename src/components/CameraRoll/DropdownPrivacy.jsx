import React, {useState} from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSortDown } from '@fortawesome/free-solid-svg-icons';
import {UpdatePhotos} from '../../services/photoServices';
import './DropdownPrivacy.css';


/** Renders DropdownPrivacy component to edit photo privacy
 * @author Khadija Khaled
 * @namespace DropdownPrivacy
 * @category Functional Component
 * @extends Component
 * @property {Array} props.imgEditIds -Array of photos ids to edit privacy
 */

function DropdownPrivacy(props) {
  const {imgEditIds} = props;
  const arrow = <FontAwesomeIcon icon={faSortDown} color="DarkGrey" />;
  const [privacy,setPrivacy] = useState("private");

  const updatePrivacy ={
    "photos":imgEditIds,
    "privacy":privacy
  }

  
  /** Updates selected photos privacy based on option selected
    * @memberof DropdownPrivacy
    * @method changePrivacy
    * @param {event} event -Tirggered event on change in input value
    */
  function changePrivacy(event){
    const privacyState=event.target.value;
    setPrivacy(privacyState);
    console.log(privacy);

    // call API
    UpdatePhotos(updatePrivacy).then( response => {
      console.log(response.status);
    })

  }



  return (
    <>

      <select onChange={changePrivacy}>
        <span className="styled" />
        <option value="private">
          Public
        </option>
        <option value="public">
          Private
        </option>
      </select>
      {arrow}

    </>
  );
}

export default DropdownPrivacy;
