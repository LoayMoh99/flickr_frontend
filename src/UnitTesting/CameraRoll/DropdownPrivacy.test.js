import React from 'react';
import ReactDOM from 'react-dom';
import DropdownPrivacy from '../../components/CameraRoll/DropdownPrivacy';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('DropdownPrivacy renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<DropdownPrivacy 
        imgEditIds="[1,2]"
        />, div);

});

it('DropdownPrivacy renders correctly enzyme', () => {
    const wrapper = shallow(<DropdownPrivacy 
        imgEditIds="[1,2]"
        />)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

