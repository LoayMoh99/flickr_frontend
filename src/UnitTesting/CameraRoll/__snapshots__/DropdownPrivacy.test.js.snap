// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownPrivacy renders correctly enzyme 1`] = `
<Fragment>
  <select
    onChange={[Function]}
  >
    <span
      className="styled"
    />
    <option
      value="private"
    >
      Public
    </option>
    <option
      value="public"
    >
      Private
    </option>
  </select>
  <FontAwesomeIcon
    border={false}
    className=""
    color="DarkGrey"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          320,
          512,
          Array [],
          "f0dd",
          "M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z",
        ],
        "iconName": "sort-down",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</Fragment>
`;
